### PR TITLE
Eslint Fixes

### PR DESCRIPTION
**Pull request by austingardner**
## Description
When we added prettier, it created some issues with eslint. We have eslint set to run when files are committed. But eslint was running on all of the files from the root directory level, ignoring the command in lint-staged to only run in a certain place. 

This PR configures eslint to run everywhere besides the ignored directories specified in the `.eslintignore` file. This seems to be a more robust strategy.

### CODE REVIEW

File: .stylelintrc-css [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": ["stylelint-config-standard", "stylelint-prettier/recommended"],
-  "ignoreFiles": ["dist/less/**/*"],
+  "ignoreFiles": ["dist/less/**/*", "**/*.min.css", "src/fonts/**/*"],
```
Comment (agliga) [inline .stylelintrc-css:3]: Why is this changed now? It was working before right?

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+.storybook
+*.min
+lint.log
+dist
+*.bundle
+scripts
+gulpfile.js
+*.bundle.js
+*.bundle.map.js
+*.stories.js
```
Comment (agliga) [inline .eslintignore:21]: I think we should check stories.js. There could be syntax or indentation problems there. 
Lets figure out how to fix that export error

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+.history
+yarn.lock
+package-lock.json
+.github
+.storybook
+*.min
+lint.log
+dist
+*.bundle
+scripts
```
Comment (agliga) [inline .eslintignore:17]: Why are we including all these files in eslint check? I thought eslint mainly checks for .js files?

File: .eslintignore [hunk 4]
```diff
@@ -0,0 +1,23 @@
+_cdn
```
Comment (ianmcburnie) [inline .eslintignore:1]: I believe we only run eslint on files under docs/src/js. Instead of doing all of these ignores, is it possible to do the inverse and just configure the path we want to run on?

Comment (austingardner) [reply .eslintignore:1]: honestly the easiest thing to do might be to run eslint on some other process and not lint-staged.

Eslint doesn't seem to work very well with lint-staged, because you're right, previously the script had eslint run on docs/src/js and it worked fine.

Comment (austingardner) [reply .eslintignore:1]: Estlint was running on the push, right?

Comment (austingardner) [reply .eslintignore:1]: We could just move the eslint script out of lint-staged and have eslint run as part of the `npm start` and `npm build` scripts

File: .eslintrc [hunk 5]
```diff
@@ -6,5 +6,8 @@
     "rules": {
         "arrow-body-style": 0,
         "prefer-arrow-callback": 0
+    },
+    "parserOptions": {
+        "sourceType": "module"
```
Comment (austingardner) [inline .eslintrc:11]: this gets rid of the error with the exports in the `.stories.js` files

File: .stylelintrc-css (hunk 1 above)
Comment (austingardner) [reply .stylelintrc-css:3]: now that lint-staged runs on committed files, when new .min.css files are generated, stylelint gets run on them; our lint:css script doesn't run on those files

Comment (agliga) [reply .stylelintrc-css:3]: What I'm wondering is when we run `build` even if min.css files are there stylelint isn't messing it up. Build does not clean anything so why is it necessary now.

Comment (austingardner) [reply .stylelintrc-css]: nuked this part

**Review by ianmcburnie — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -29,7 +29,7 @@
         "lint": "npm run lint:less && npm run lint:css && npm run lint:js",
         "lint:less": "stylelint docs/src/less src/less --config .stylelintrc-less",
         "lint:css": "stylelint dist --config .stylelintrc-css",
-        "lint:js": "eslint 'docs/src/js/base/*.js' > lint.log",
+        "lint:js": "eslint 'docs/src/js/base/*.js' > lint.log && eslint 'src/less/**/stories/*.js' > lint.log ",
```
Comment (agliga) [inline package.json:32]: Only quick nit...
By doing it this way, it will override the lint.log file for the second time and won't keep the data from the first. 
You should change it to this:
`eslint 'docs/src/js/base/*.js' > lint.log && eslint 'src/less/**/stories/*.js' >> lint.log` (double `>>` appends to the previous one, while `>` clears the log file and then appends)

**Review by agliga — COMMENTED** [review]
One minor nit

File: package.json (hunk 6 above)
Comment (austingardner) [reply package.json]: good catch, thanks! Fixed

**Review by agliga — APPROVED** [review]
